### PR TITLE
ISPN-5714 Add value/metadata aware store expiration listener

### PR DESCRIPTION
**Pull request by wburns**
https://issues.jboss.org/browse/ISPN-5714

The only store I converted over was the DummyInMemoryStore as it is for tests.  I didn't want to adversely affect performance until we can verify it is alright or not.

### CODE REVIEW

**Comment by wburns** [comment]
1 sec fixing an issue real quick.

**Comment by wburns** [comment]
Alright should be good now

File: core/src/test/java/org/infinispan/expiration/impl/ExpirationStoreListenerFunctionalTest.java [hunk 1]
```diff
@@ -62,8 +62,8 @@ public void testExpirationOfStoreWhenDataNotInMemory() throws Exception {
       assertFalse(event.isPre());
       assertNotNull(event.getKey());
       // Unfortunately we don't guarantee value or metadata from store
```
Comment (danberindei) [inline core/src/test/java/org/infinispan/expiration/impl/ExpirationStoreListenerFunctionalTest.java]: Comment is now outdated?

Comment (wburns) [inline core/src/test/java/org/infinispan/expiration/impl/ExpirationStoreListenerFunctionalTest.java]: Sure I can change it

**Comment by danberindei** [comment]
Integrated, thanks Will!

**Pull request closed without merging**